File: tests/integration/test_embeddings_endpoint.py
# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
# SPDX-License-Identifier: Apache-2.0
"""Tests for /v1/embeddings endpoint."""

import pytest

from tests.integration.conftest import AIPerfCLI
from tests.integration.models import AIPerfMockServer


@pytest.mark.integration
@pytest.mark.asyncio
class TestEmbeddingsEndpoint:
    """Tests for /v1/embeddings endpoint."""

    async def test_basic_embeddings(
        self, cli: AIPerfCLI, aiperf_mock_server: AIPerfMockServer
    ):
        """Basic embeddings request."""
        result = await cli.run(
            f"""
            aiperf profile \
                --model nomic-ai/nomic-embed-text-v1.5 \
                --tokenizer gpt2 \
                --url {aiperf_mock_server.url} \
                --endpoint-type embeddings \
                --request-count 10 \
                --concurrency 2 \
                --ui simple
            """
        )
        assert result.request_count == 10
        assert (
            not hasattr(result.json, "time_to_first_token")
            or result.json.time_to_first_token is None
        )

File: rust/integration/tests/test_embeddings_endpoint.rs
// SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
// SPDX-License-Identifier: Apache-2.0
#[path = "common/mod.rs"]
mod common;
use common::*;

// Tests for /v1/embeddings endpoint.

/// Basic embeddings request completes with expected request count.
#[tokio::test]
async fn test_basic_embeddings() {
    let h = AIPerfHarness::new().await;
    let r = h.run(&format!(
        "--model nomic-ai/nomic-embed-text-v1.5 \
         --url {} \
         --endpoint-type embeddings \
         --request-count {DEFAULT_REQUEST_COUNT} \
         --concurrency {DEFAULT_CONCURRENCY} \
         --workers-max {DEFAULT_CONCURRENCY} \
         --ui simple",
        h.mock.url
    ));
    assert_eq!(r.artifacts.request_count() as u32, DEFAULT_REQUEST_COUNT);
    // Embeddings are non-streaming, so streaming metrics should not be present.
    assert!(
        r.artifacts
            .json()
            .get("time_to_first_token")
            .map_or(true, |v| v.is_null())
    );
}
